components/display/schema.json

```diff
@@ -0,0 +1,88 @@
+    "type": "object",
+    "required": [
+        "displayName",
+        "vendor"
+    ],
+    "additionalProperties": false,
+    "properties": {
+        "displayName": {
+            "description": "The human-friendly name of this component.",
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 30
+        },
+        "published": {
+            "description": "If true, this component is supported by the current firmware version and will be displayed to all users. If false, it is hidden behind a developer toggle so that contributors can still work on it against the production site.",
+            "type": "boolean"
+        },
+        "description": {
+            "description": "A brief description describing this component's capabilities.",
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 255
+        },
+        "productURL": {
+            "description": "Link to this component's homepage.",
+            "type": "string",
+            "format": "uri"
+        },
+        "documentationURL": {
+            "description": "Link to this component's documentation.",
+            "type": "string",
+            "format": "uri"
+        },
+        "vendor": {
+            "description": "Name of the company that makes this component.",
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 24
+        },
+        "displayType": {
+            "description": "Display configuration and type, maps to DisplayAddOrReplace",
+            "type": "object",
+            "required": [
+                "type"
+            ],
+            "properties": {
+                "type": {
+                    "description": "The type of display",
+                    "type": "string",
+                    "enum": [
+                        "unspecified",
+                        "oled",
+                        "epd"
+                    ]
+                },
+                "spiEpd": {
+                    "description": "Optional EPD SPI configuration.",
+                    "type": "object",
+                    "properties": {
+                        "bus": {
+                            "type": "integer"
+                        }
+                    }
+                },
+                "epdConfig": {
+                    "description": "Optional EPD Configuration",
+                    "type": "object",
+                    "required": [
+                        "mode"
+                    ],
+                    "properties": {
+                        "mode": {
+                            "type": "string",
+                            "enum": [
+                                "unspecified",
+                                "grayscale4",
+                                "mono"
+                            ]
+                        }
+                    }
+                }
```

Comment:
Should we have the option to configure epd reset pins etc here?
i.e. when we add the featherwing EPD of the same type/size as magtag2025 we'll need to define the appropriate pins via "labels".

I guess that's blocked by us not supporting defining more than one pin per component (like why RGB pixel is broken).

Reply:
There is no concept of pins in the component definitions and schemas, only the pin label hint name for some component types (string matching suggestion for board pins like neopixel).
That's why I mentioned the RGB component which requires 3pins and cannot currently be supported, so I guess for v1 schema we cannot / will not support multiple pins (or any pins) being defined in component definitions.

Reply:
Yeah this will have to be fore the v2 impl. of display component.